tests/run-tests

```diff
@@ -189,7 +189,12 @@ def run_micropython(pyb, args, test_file):
 
     return output_mupy
 
-def run_tests(pyb, tests, args):
+
+def run_feature_check(pyb, args, test_file):
```

Comment:
Ok, done. Generally, my feeling is that while we add useful features to it, run-tests gets hairier and hairier. So, any ideas on making the changes to be less so are welcome ;-).